infrastructure/code/template/details.html

```diff
@@ -824,18 +855,24 @@ <h2 class="data-flow-header">
                 Data Flow - {{len .DataFlow}} {{if gt (len .DataFlow) 1}}steps{{else}}step{{end}}
               </h2>
               <div class="data-flow-body">
-                <table class="data-flow-body">
+                <table class="data-flow-table">
                   <tbody>
-                    {{range .DataFlowTable}}
-                    <tr class="data-flow-row">
-                      <td class="data-flow-number">{{.Number}}</td>
-                      <td class="data-flow-clickable-row" file-path="{{.FilePath}}" start-line="{{.StartLine}}"
-                        end-line="{{.EndLine}}" start-character="{{.StartCharacter}}" end-character="{{.EndCharacter}}">
-                        {{.FileName}}:{{.StartLineValue}}
-                      </td>
-                      <td class="data-flow-delimiter">|</td>
-                      <td class="data-flow-text">{{.Content}}</td>
-                    </tr>
+                    {{range $fileName, $dataFlowItems := .DataFlowTable}}
+                    <div class="data-flow-file">
+                      {{$.FileIcon}}
+                      <span class="data-flow-file-name">{{$fileName}}</span>
+                    </div>
+                    {{range $dataFlowItems}}
+                      <tr class="data-flow-row">
+                        <td class="data-flow-number">{{.Number}}</td>
+                        <td class="data-flow-clickable-row" file-path="{{.FilePath}}" start-line="{{.StartLine}}"
+                          end-line="{{.EndLine}}" start-character="{{.StartCharacter}}" end-character="{{.EndCharacter}}">
+                          line:{{.StartLineValue}}
+                        </td>
+                        <td class="data-flow-delimiter">|</td>
+                        <td class="data-flow-text">{{.Content}}</td>
+                      </tr>
+                    {{end}}
```

Comment:
I replicated two folders from Juice Shop and it seems that we are not placing the file names correctly between blocks:

- Figma design:

![Data Flow Figma](https://github.com/snyk/snyk-ls/assets/1948377/80a1f4e0-c646-4da1-a6ff-14342612fce9)

- IntelliJ
<img width="1280" alt="image" src="https://github.com/snyk/snyk-ls/assets/1948377/a95cce84-c717-4d6e-92ae-9da92e5b43fb">

I pushed this folders to the testing repo I shared with you :)